packages/compiler/core/types.ts

```diff
@@ -1252,18 +1253,20 @@ export interface CadlLibraryDef<
    * Provide names for emitters if there is multiple.
    */
   readonly emitter?: {
-    names?: readonly E[];
+    options?: JSONSchemaType<E>;
```

Comment:
Currently the options are a JSon schema which lets the options be a complex object with complex validation
![image](https://user-images.githubusercontent.com/1031227/167949844-c40d2866-3f3a-4dc5-93cf-2d1eb2b53482.png)

Should we stick to have that and having to use `ajv` to validate the options or should make that a much simpler system which only allows `key-> value` with a simple primitive validation for the value

Reply:
Good question. This seems nice, but my only hesitation is if there are any consequences to pulling in this dependency on JSONSchemaType in in core types. @beterlson for thoughts.

Reply:
How does the validation work if I have an emitter that needs some custom options?

Reply:
https://github.com/microsoft/cadl/blob/07d376074de4480ce0a58ff7b892369328e30e3d/packages/openapi3/src/lib.ts#L7-L14
This is how an emitter defines the options. All emitter options are "custom". Was that the question?

Reply:
I am not following how this looks on the command line, does it mean we can provide json for emitter options on the command line, that we specify a configuration file for the emitter optiosn on the command line, or something else?

Reply:
But right now we can set emitter options with `--option key=value` will this be removed when we go to a more complex emitter option schema?

Reply:
I like this, this keeps options between emitters from clashing and makes the command very explicit.  I think, from a  usability perspective, we will want to make it possible to set emitter-specific options from the command line similar to this.

We should also look through the documentation to see where we may need to update the docs for the new command line option setting

Reply:
But we will also want the command line help message to make this easy for users to discover, probably with an example (let's say from the OpenAPI3 emitter)